synapse/axon.py

```diff
@@ -73,24 +70,85 @@ def __init__(self, datadir, **opts):
             self._fireAxonIden(iden)
 
         # fire auto-run axons
-        auto = self.opts.get('autorun')
-        while len(self.axons) < auto:
+        auto = self.getConfOpt('axonhost:autorun')
+        while (len(self.axons) - len(self.cloneaxons)) < auto:
             self.add()
 
+        url = self.getConfOpt('axonhost:axonbus')
+        if url:
+            self.axonbus = s_service.openurl(url)
+            self.axonbus.runSynSvc(s_common.guid(), self)
+
+    @staticmethod
+    @s_config.confdef(name='axonhost')
+    def _axonhost_confdefs():
+        confdefs = (
+            ('axonhost:autorun', {'type': 'int', 'defval': 0,
+                                  'doc': 'Number of Axons to autostart.'}),
+            ('axonhost:axonbus', {'type': 'str', 'defval': '',
+                                  'doc': 'URL to an axonbus'}),
+            ('axonhost:bytemax', {'type': 'int', 'defval': terabyte,
+                                  'doc': 'Max size of each axon created by the host.'}),
+            ('axonhost:clones', {'type': 'int', 'defval': 2,
+                                 'doc': 'The default number of clones for a axon.'}),
+            ('axonhost:maxsize', {'type': 'int', 'defval': 0,
+                                  'doc': 'Max total allocations for Axons created by the host. '
+                                         'Only applies if set to a positive integer.'}),
+            ('axonhost:syncmax', {'type': 'int', 'defval': gigabyte * 10,
+                                  'doc': ''}),  # XXX Words
+            ('axonhost:hostname', {'type': 'str', 'defval': s_thishost.get('hostname'),
+                                   'doc': 'AxonHost hostname'}),
+        )
+        return confdefs
+
     def _fireAxonIden(self, iden):
         axondir = s_common.gendir(self.datadir, '%s.axon' % iden)
 
-        opts = dict(self.opts)
+        opts = self.makeAxonOpts()
         jsopts = s_common.jsload(axondir, 'axon.opts')
         if jsopts is not None:
             opts.update(jsopts)
 
+        # Special case where the axonbus may update - we want to ensure
+        # we're passing the latest axonbus to the Axon so it can register
+        # itself properly.
+        if 'axon:axonbus' in opts:
+            axonbus = opts.get('axon:axonbus')
```

Comment:
This construct causes 2 hash calculations and tree lookups.  Most of the rest of the code base does this like so:
```
axonbus = opts.get('axonbus')
if axonbus is not None:
    dostuff()
```

Reply:
fixed